### PR TITLE
Issue #4770 - Deprecate GzipTester

### LINKED ISSUE #4770: Simplify Gzip testing
The old `GzipTester` is getting long in the tooth and needs to be simplified and ideally removed entirely.

The Gzip based tests in general can benefit from having more local code to understand what's going on.

### PR DESCRIPTION
**Pull request by joakime**
+ Renamed TestDirContentServlet to AbstractFileContentServlet
+ Renamed various TestServlet*Write classes to BlockingServlet*Write to better
  represent their purpose.
+ Renamed TestServletBufferTypeLengthWrite to HttpOutputWriteFileContentServlet
+ Cleaned up GzipContentLengthTest to be one big ParameterizedTest
  - now does Internal vs External GzipHandler testing too.
+ Introduced AbstractGzipTest to house simple common methods
+ Cleanup remaining testcases

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

**Comment by sbordet** [comment]
@joakime a ton of test failures, can you please check?

**Comment by joakime** [comment]
taking a look

**Comment by joakime** [comment]
@sbordet the build is clean on this PR now

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/AbstractGzipTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+public abstract class AbstractGzipTest
+{
+    protected static final int DEFAULT_OUTPUT_BUFFER_SIZE;
+
+    static
+    {
+        HttpConfiguration defaultHttp = new HttpConfiguration();
+        DEFAULT_OUTPUT_BUFFER_SIZE = defaultHttp.getOutputBufferSize();
+    }
```
Comment (sbordet) [inline jetty-servlets/src/test/java/org/eclipse/jetty/servlets/AbstractGzipTest.java]: Remove `static`  block -- just do `DEFAULT_OUTPUT_BUFFER_SIZE = new HttpConfiguration().getOutputBufferSize()`

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultServletDeferredContentTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            {
+                // System.err.println("type for "+uri.substring(0,uri.length()-9)+" is "+getServletContext().getMimeType(uri.substring(0,uri.length()-9)));
+                resp.setContentType(getServletContext().getMimeType(uri.substring(0, uri.length() - 9)));
+            }
+
+            doGet(req, resp);
+        }
+    }
+
+     */
```
Comment (sbordet) [inline jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultServletDeferredContentTypeTest.java]: Why it's all commented out?

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipContentLengthTest.java [hunk 3]
```diff
@@ -42,317 +49,180 @@
...
-            return String.format("%s [%,d bytes, compressed=%b]", fileName, fileSize, expectCompressed);
+            contentEncodingMatcher = not(contentEncodingMatcher);
         }
+        assertThat("Content-Encoding", response.get("Content-Encoding"), contentEncodingMatcher);
+
+        // Response Content checks
+        UncompressedMetadata metadata = parseResponseContent(response);
+        assertThat("(Uncompressed) Content Length", metadata.uncompressedSize, is(fileSize));
+        // TODO: Assertions.assertArrayEquals();
+        // TODO: compare bytes
```
Comment (sbordet) [inline jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipContentLengthTest.java]: Fix TODOs?

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultServletDeferredContentTypeTest.java (hunk 2 above)
Comment (joakime) [reply jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipDefaultServletDeferredContentTypeTest.java]: I need to fix this one.

File: jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipContentLengthTest.java (hunk 3 above)
Comment (joakime) [reply jetty-servlets/src/test/java/org/eclipse/jetty/servlets/GzipContentLengthTest.java]: I'll address these

**Comment by joakime** [comment]
@sbordet Updated 3 test classes based on your review.
This branch / PR is now green on CI.

**Review by sbordet — APPROVED** [review]

**Pull request merged**